pwiz_tools/Skyline/FileUI/PeptideSearch/BuildPeptideSearchLibraryControl.cs

```diff
@@ -643,5 +656,15 @@ private void comboInputFileType_SelectedIndexChanged(object sender, EventArgs e)
         {
             UpdatePerformDDASearch();
         }
+
+        #region test_support
+
+        // For test purposes, lets us test error handling
+        public void SetCutoffControlText(string cutoffControlText)

```

Comment:
If properties like CutOffScore are only supposed to be used by tests and not by other Skyline application code, then I wonder whether it should be a string that gets/sets textCutoff.Text. Other than that, PR looks good.

Reply:
I don't understand. The user can only deal in strings, we have to convert strings like textCutoff to values like CutoffScore, and gracefully deal with users typing in strings that don't make sense. We're pretty good about that, but for whatever reason this particular control never got any such validation code. The problem has been lurking for at least ten years.

Reply:
I see. Keep in mind that the test would have to think about L10N when trying to represent fractional values, though. Probably even more awkward.

Reply:
Yes, devs would have to remember to do the `.ToString(CultureInfo.CurrentCulture)` bit after the number (or use a helper function). Which I admit is easy to forget (and I have done so).

However the `double CutoffScore` is much closer to the way the rest of Skyline does this testing. The difference is things like transition settings UI use a nullable property like `double? CutoffScore` and test that when setting CutoffScore to null, it triggers the `Resources.MessageBoxHelper_ValidateDecimalTextBox__0__must_contain_a_decimal_value` error. Is there a reason not to use `ControlUtil.ValidateDecimalTextBox` here? Testing a null control value is probably close enough to an invalid value like "12%@$@": they both should generate the same error.

Reply:
Yes, I should use that more centralized code for consistency, will do.